cbits/posix/runProcess.c

```diff
@@ -21,21 +24,25 @@
 #include <signal.h>
 #endif
 
-int
-get_max_fd()
-{
-    static int cache = 0;
-    if (cache == 0) {
-#if HAVE_SYSCONF
-        cache = sysconf(_SC_OPEN_MAX);
-        if (cache == -1) {
-            cache = 256;
+#define CLOSE_RANGE_SYSCALL_NUMBER 436
+
+void
+closefrom_excluding(int lowfd, int excludingFd) {
+    // Try using the close_range syscall, provided in Linux kernel >= 5.9.
+    // We do this directly because not all C libs provide a wrapper (like musl)
+    long ret = syscall(CLOSE_RANGE_SYSCALL_NUMBER, lowfd, excludingFd - 1, 0);
```

Comment:
This line is a little dodgy due to the direct syscall. It should be good on Linux kernels because Linux has very strong guarantees about syscall stability. For other POSIX platforms we probably need to gate this behind a macro. Another option would be to not even try using the `close_range` syscall, and just loop from `lowFd` to `excludingFd`, assuming that `excludingFd` will be reasonably small.

FWIW there's a possibility that `close_range` will be added to `musl` in the future: https://inbox.vuxu.org/musl/20220810130311.dwk7zxwkz32igrdm@yandex.com/T/#u

It would help me move this along if I had an easy way to test `musl` builds. I have a somewhat convoluted method using Haskell.nix, but do you happen to know a good way @bgamari ?

Reply:
I would likely just use GHC's CI infrastructure. That is, open a draft MR against [`ghc/ghc`](https://gitlab.haskell.org/ghc/ghc) bumping the `process` submodule to your branch. This will get tested against Alpine 3.12.